iroha_logger/src/telemetry.rs

```diff
@@ -0,0 +1,142 @@
+        self.fields
+            .push((field.name(), format!("{:?}", &value).into()))
+    }
+    fn record_i64(&mut self, field: &Field, value: i64) {
+        self.fields.push((field.name(), value.into()))
+    }
+    fn record_u64(&mut self, field: &Field, value: u64) {
+        self.fields.push((field.name(), value.into()))
+    }
+    fn record_bool(&mut self, field: &Field, value: bool) {
+        self.fields.push((field.name(), value.into()))
+    }
+    fn record_str(&mut self, field: &Field, value: &str) {
+        self.fields.push((field.name(), value.into()))
+    }
+    fn record_error(&mut self, field: &Field, mut error: &(dyn Error + 'static)) {
+        let mut vec = vec![error.to_string()];
+        while let Some(inner) = error.source() {
+            error = inner;
+            vec.push(inner.to_string())
+        }
+        self.fields.push((field.name(), vec.into()))
+    }
+}
+
+impl Telemetry {
+    fn from_event(target: &'static str, event: &Event<'_>) -> Self {
+        let fields = TelemetryFields::default();
+        let mut telemetry = Self { target, fields };
+        event.record(&mut telemetry);
+        telemetry
+    }
+}
+
+/// Telemetry layer
+#[derive(Debug, Clone)]
+pub struct TelemetryLayer<S: Subscriber> {
+    telemetry_sender: Sender<Telemetry>,
+    subscriber: S,
+}
+
+impl<S: Subscriber> TelemetryLayer<S> {
+    /// Create telemetry from channel sender
+    pub fn from_sender(subscriber: S, telemetry_sender: Sender<Telemetry>) -> impl Subscriber {
+        EventSubscriber(Self {
+            subscriber,
+            telemetry_sender,
+        })
+    }
+
+    /// Create new telemetry layer with specific channel size (via const generic)
+    #[allow(clippy::new_ret_no_self)]
+    pub fn new<const CHANNEL_SIZE: usize>(subscriber: S) -> (impl Subscriber, Receiver<Telemetry>) {
+        let (sender, reciever) = sync::channel(CHANNEL_SIZE);
+        let telemetry = Self::from_sender(subscriber, sender);
+        (telemetry, reciever)
+    }
+
+    /// Create new telemetry layer with specific channel size
+    #[allow(clippy::new_ret_no_self)]
+    pub fn from_capacity(
+        subscriber: S,
+        channel_size: usize,
+    ) -> (impl Subscriber, Receiver<Telemetry>) {
+        let (sender, reciever) = sync::channel(channel_size);
+        let telemetry = Self::from_sender(subscriber, sender);
+        (telemetry, reciever)
+    }
+}
+
+impl<S: Subscriber> EventInspectorTrait for TelemetryLayer<S> {
+    type Subscriber = S;
+    fn inner_subscriber(&self) -> &Self::Subscriber {
+        &self.subscriber
+    }
+    fn event(&self, event: &Event<'_>) {
+        let target = event.metadata().target();
+        if target.starts_with("telemetry::") {
+            #[allow(clippy::unwrap_used)]
+            let target = target.strip_prefix("telemetry::").unwrap();
+            let _ = self
+                .telemetry_sender
+                .try_send(Telemetry::from_event(target, event));
+        } else {
```

Comment:
```suggestion
        if let Ok(target) = target.strip_prefix("telemetry::") {
            let _ = self
                .telemetry_sender
                .try_send(Telemetry::from_event(target, event));
        } else {
```

Reply:
Also please move string into const.

Reply:
If let construct doesnt show that this behaviour is expected and will always be true

Reply:
You still check with `starts_with`, so I just combined unwrap and this check, what you do here is basically the same as checking is_ok and then unwrapping.